pkg/configuration/parameters_test.go

```diff
@@ -1,317 +1,104 @@
-		Username:         &fakePassword,
-		Password:         &fakePassword,
-		IssueList:        make([]string, 1),
-		Context:          EditCustomField,
-		CustomFieldValue: &fakeCustomFieldValue,
-	}
-
-	param.validate()
-	testExpectedBoolResult(t, param.Meta.valid, true)
-}
+type TestJiraAPIResourceParameters struct {
+	JiraAPIUrl       string   `faker:"url"`
+	Username         string   `faker:"username"`
+	Password         string   `faker:"password"`
+	Destination      string   `faker:"word"`
+	IssueList        []string `faker:"len=5"`
+	ClosedStatusName string   `faker:"word"`
+	TransitionName   string   `faker:"word"`
+	Context          configuration.Context
 
-func TestContextEditCustomFieldSuccess2(t *testing.T) {
-	param := &JiraAPIResourceParameters{
-		JiraAPIUrl:               &fakeUrl,
-		Username:                 &fakePassword,
-		Password:                 &fakePassword,
-		IssueList:                make([]string, 1),
-		Context:                  EditCustomField,
-		CustomFieldValueFromFile: &fakeCustomFieldValue,
-	}
+	EditCustomFieldParam TestJiraApiResourceParametersEditCustomField
+	AddComment           JiraApiResourceParametersAddComment
 
-	param.validate()
-	testExpectedBoolResult(t, param.Meta.valid, true)
+	ActiveIssue string
 }
 
-func TestContextEditCustomFieldFailMissingBothValuesNil(t *testing.T) {
-	param := &JiraAPIResourceParameters{
-		JiraAPIUrl: &fakeUrl,
-		Username:   &fakePassword,
-		Password:   &fakePassword,
-		IssueList:  make([]string, 1),
-		Context:    EditCustomField,
-	}
-
-	param.validate()
-	testExpectedBoolResult(t, param.Meta.valid, false)
-
-	if param.Meta.Msg == "" {
-		t.Errorf("String value was incorrect, got empty string but expected a message")
-	}
-}
-
-func TestContextEditCustomFieldFailMissingBothValuesEmpty(t *testing.T) {
-	param := &JiraAPIResourceParameters{
-		JiraAPIUrl:               &fakeUrl,
-		Username:                 &fakePassword,
-		Password:                 &fakePassword,
-		IssueList:                make([]string, 1),
-		Context:                  EditCustomField,
-		CustomFieldValue:         &emptyValue,
-		CustomFieldValueFromFile: &emptyValue,
-	}
-
-	param.validate()
-	testExpectedBoolResult(t, param.Meta.valid, false)
-
-	if param.Meta.Msg == "" {
-		t.Errorf("String value was incorrect, got empty string but expected a message")
-	}
+type TestJiraApiResourceParametersEditCustomField struct {
+	CustomFieldName          string `faker:"word"`
+	CustomFieldType          string `faker:"word"`
+	CustomFieldValue         string `faker:"word"`
+	CustomFieldValueFromFile string `faker:"word"`
 }
 
-func TestInitializeIssueListSingle1(t *testing.T) {
-	param := &JiraAPIResourceParameters{}
-
-	list := "ABC-001"
-	param.initializeIssueList(&list)
-
-	if param.IssueList == nil {
-		t.Errorf("IssueList was nil but should have contained at least one element")
-	}
-
-	testExpectedBoolResult(t, param.Meta.MultipleIssue, false)
+type JiraApiResourceParametersAddComment struct {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [type JiraApiResourceParametersAddComment should be JiraAPIResourceParametersAddComment](https://app.codacy.com/manual/TurnsCoffeeIntoScripts/jira-api-resource/pullRequest?prid=5035521)